### PR TITLE
PPF-188-support slack

### PR DESCRIPTION
**Pull request by vhande**
### Added
- SupportController
- InformationDialog

---
Ticket: https://jira.uitdatabank.be/browse/PPF-188

### CODE REVIEW

File: app/Http/Controllers/SupportController.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            'email' => Auth::user()?->email,
+            'slackSuccess' => $slackSuccess,
+            'slackError' => $slackError,
+        ]);
+    }
+    public function sendInvitation(Request $request): RedirectResponse
+    {
+        $botToken = config('slack.botToken');
+        $channelID = config('slack.channelID');
+        $email = Auth::user()?->email;
```
Comment (simon-debruijn) [inline app/Http/Controllers/SupportController.php:33]: The `sendInvitation` method should probably be under the authenticated routes. If the user is not authenticated and on the support page and clicks the join slack button we first let them login.

Then we'll be sure that `Auth::user()` is not null

File: resources/translations/nl.json [hunk 2]
```diff
@@ -259,7 +259,11 @@
   },
   "dialog": {
     "confirm": "Bevestigen",
-    "cancel": "Annuleren"
+    "cancel": "Annuleren",
+    "close": "Sluiten",
+    "invite_success": "Wij hebben een uitnodiging gestuurd naar {{email}}",
+    "invite_wrong": "Er is iets fout gegaan. Probeer het later opnieuw."
```
Comment (simon-debruijn) [inline resources/translations/nl.json:265]:
```suggestion
    "invite_error": "Er is iets fout gegaan. Probeer het later opnieuw."
```
For consistency

File: routes/web.php [hunk 3]
```diff
@@ -38,9 +38,12 @@
         '/en/support',
         '/nl/ondersteuning',
     ],
-    static fn () => Inertia::render('Support/Index')
+    [SupportController::class, 'index'],
+    'support.index'
 );
 
+Route::post('/support/slack', [SupportController::class, 'sendInvitation']);
```
Comment (simon-debruijn) [inline routes/web.php:45]:
```suggestion
Route::post('/support/slack/invite', [SupportController::class, 'sendSlackInvitation']);
```

**Review by simon-debruijn — CHANGES_REQUESTED** [review]

File: resources/ts/Components/SupportCard.tsx [hunk 4]
```diff
@@ -4,16 +4,31 @@ import { Heading } from "./Heading";
...
   actionTitle,
   actionUrl,
+  email,
+  slackError,
+  slackSuccess,
 }: Props) => {
+  const { t } = useTranslation();
+  const translateRoute = useTranslateRoute();
+  const handleSlackInvitation = () => router.post("/support/slack");
+  const handleRedirect = () => router.get(translateRoute("/support"));
```
[the remark this answers is not in the document]
Comment (vhande) [reply resources/ts/Components/SupportCard.tsx:30]: It doesn't work as we have one more redirect after this.

File: app/Http/Controllers/SupportController.php [hunk 5]
```diff
@@ -39,7 +39,13 @@ public function sendInvitation(Request $request): RedirectResponse
...
                 ]);
 
-            if (!$response->ok()) {
+            $body = json_decode(
+                json: $response->body(),
+                associative: true,
+                flags: JSON_THROW_ON_ERROR
+            );
+
+            if (!$response->ok() | $body['ok'] === false) {
```
Comment (simon-debruijn) [inline app/Http/Controllers/SupportController.php:48]:
```suggestion
            if (!$response->ok() || $body['ok'] === false) {
```

**Review by simon-debruijn — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/SupportController.php [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        $botToken = config('slack.botToken');
+        $channelID = config('slack.channelID');
+        $email = Auth::user()?->email;
+
+        try {
+            $response = Http::withToken($botToken)
+                ->post('https://slack.com/api/conversations.inviteShared', [
+                    'channel' => $channelID,
+                    'emails' => $email,
```
Comment (simon-debruijn) [inline app/Http/Controllers/SupportController.php:39]: I now see in the slack docs that emails is supposed to be an array. So even if this works, it is better to follow the correct guidelines.

```suggestion
                    'emails' => [$email],
```

File: app/Http/Controllers/SupportController.php [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+{
+    public function index(Request $request): Response
+    {
+        $slackSuccess = $request->query->has('slackSuccess');
+        $slackError = $request->query->has('slackError');
+
+        return Inertia::render('Support/Index', [
+            'email' => Auth::user()?->email,
+            'slackSuccess' => $slackSuccess,
+            'slackError' => $slackError,
```
Comment (simon-debruijn) [inline app/Http/Controllers/SupportController.php:26]: We can also merge these into 1 `slackStatus` which can be `success` or `error`

**Review by simon-debruijn — APPROVED** [review]

**Pull request merged**